api/datasets.go

```diff
@@ -0,0 +1,42 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+func (c *CodeListAPI) getCodeDatasets(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	vars := mux.Vars(r)
+	id := vars["id"]
```

Comment:
Somewhat subjective - consider being more explicit here and using `codeListID` instead of `id`